Phase1 Pixel DQM Histogram Reorganisation

schneiml opened the pull request:
This pull request starts to rearrange the Phase1 plots closer to their final layout:

- Different quantities use different top folders
- The last level of grouping is reflected in the name, not in the folder, to have all 'same level' plots next to each other.
- DetIds on the lowest level are replaced by online module names.
- `nclusters` redefined to count only size>1 clusters, since this is the significant quantity for e.g. timing scans.

There are some more or less unrelated changes to `SiPixelPhase1Common`:
- The ability to use `|` in Specifications to separate FPIX/BPIX names has been removed, instead FPIX and BPIX Specifications are now independent. This simpilfies the control flow in the `HistogramManager` a bit and removes some hard-to-understand behaviour. The downside is that data may now be silently dropped if the Specification does not cover the full detector (`bookUndefined = False`), since it is no longer possible to cover the full detector at once.

- The naming changes mentioned above required refactoring the histogram naming code significantly, things are now a bit more complicated but consistent.

- `GeometryInterface` now uses floating-point instead of integer values, which should allow better automatic labels on some plots.

- A simple plugin (`SiPixelPhase1GeometryDebug`) outputs various Geometry quantities plotted against each other, to check if things come out as expected.

And finally some changes to the online clients. They now use the file input source and the `pixel` client is phase1 only, even though it is still configured for phase0 data. A variant using phase1 data exists in `DQM/SiPixelPhase1Config/test`.

The histogram layout in the `PixelPhase1/` folder is still far from final, and will take some more time.

(also @fioriNTU @davidcarbonis )

schneiml committed:
Add a small Plugin that plots geometry related quantities against geometry.

fioriNTU committed:
minor modifications to Phase 1 folders structure

schneiml committed:
Improve handing of (readout) channels.

schneiml committed:
Allow fractional values in the GeometryInterface, custom binning.

schneiml committed:
Move all plots one level up.

The last grouping is appended to the name. This allows having e.g. the 4 Layers next to each other, instead of in 4 subfolders.

Required a bit of refactoring. Be carful, grouping by just DetId can give you thousands of plots in one folder, which the GUI might not like.

schneiml committed:
Clean up FED stuff a bit.

This uses a side effect of |-columns, wwe should find a better way.

schneiml committed:
Simplify passing predefined Specs.

This uses a Python-Level hack to flatten lists, so that a list of Specs can be passed like a single Spec.

schneiml committed:
Remove usages of |-columns.

This involves duplicating all Specs and separating Barrel and Forward parts.

schneiml committed:
Remove the debug plot from the Digis plugin, it is now independent.

schneiml committed:
Remove the or-column functionality.

Actually, it turn out that a lot of code removed here was not really needed even with or-columns; so they could be added back in with little effort

schneiml committed:
Fix Raw Data and harvesting after all these changes

schneiml committed:
Do names properly, such that it also works for Harvesting

schneiml committed:
Move formatValue to the GeometryInterface, where it actually belongs.

schneiml committed:
Add pretty module names.

Also might improve booking/harvesting performance slightly (memoization) and fixes some layout issues.

schneiml committed:
Add some nclusters histos and disable debug plots by default.

schneiml committed:
Rearrange tracking-related plots into the Tracks/ folder.

This also moves the cluserprob histos into a better fitting plugin.

cmsbuild commented:
A new Pull Request was created by  @schneiml (Marcel Schneider) for CMSSW_9_0_X.

It involves the following packages:

DQM/SiPixelPhase1Clusters
DQM/SiPixelPhase1Common
DQM/SiPixelPhase1Config
DQM/SiPixelPhase1Digis
DQM/SiPixelPhase1RawData
DQM/SiPixelPhase1RecHits
DQM/SiPixelPhase1TrackClusters
DQM/SiPixelPhase1TrackEfficiency
DQM/SiPixelPhase1TrackResiduals


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@idebruyn, @threus, @fioriNTU, @hdelanno this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

schneiml committed:
Set up PB DQM for using file input and adopt latest changes

schneiml committed:
Cut in cluster size > 1 for nclusters.

schneiml committed:
Update the production online pixel DQM.

This completely disables Phase0 DQM there, but the data remains Phase0.
Switched to file input and lxplus golbaltag.
This can be used for online development and merged with the one in SiPixelPhase1Config/testfor production.
Also needs to be switched back to Streamer input and P5 cluster conditions for Productions, but these settings cannot be tested currently.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_9_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: 275cf14de66c4e7b23299349b8a5d55b9026a14f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16875/16968/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16875/16968/summary.html

davidlange6 commented:
+1
